lib/mc_translator.rb

```diff
@@ -134,31 +134,41 @@ def push(files)
 
     def pull(files)
       files
-        .map do |file| file[:path] end
+        .map { |file| file[:path] }
         .each do |file|
           config['locales'].each do |locale|
-            expansion = {
-              locale: locale,
-              dir: File.dirname(file),
-              name: File.basename(file, '.*').gsub(/en-US/, ''),
-              ext: File.extname(file),
-            }
-
-            name = File.expand_path(config['rewrite'] % expansion)
-            content = @files.download_translated(
-                        file,
-                        locale,
-                        { retrievalType: 'published' }
-                      )
-
-            p file
-            p name
-
-            dir = File.dirname(name)
-            Dir.mkdir(dir) unless Dir.exist?(dir)
-            File.write(name, content)
+            pull_with_locale(file, locale, locale, 'published')
           end
+
+          pull_with_locale(file, 'en-US', 'mc-US', 'pseudo')
         end
     end
+
+    def pull_with_locale(file, locale_on_smartling, locale_on_mc, type)
+      expansion = {
+        locale: locale_on_mc,
+        dir: File.dirname(file),
+        name: File.basename(file, '.*').gsub(/en-US/, ''),
+        ext: File.extname(file),
+      }
+
+      name = File.expand_path(config['rewrite'] % expansion)
+      content = @files.download_translated(
+        file,
+        locale_on_smartling,
+        retrievalType: type
+      )
+
+      p file
+      p name
+
+      if locale_on_smartling != locale_on_mc && /\.yml/ =~ name
```

Comment:
because i'm not super pro at `ruby`, can you walk me through what this is doing?

Reply:
Sure, first it checks if the locale that Smartling uses is the same we use, this is because we save pseudo locale as `mc-US` but they have it as `en-US`. Then it checks if the name of the file has `.yml` in it. 

What I noticed happens is that the yml files have a key with the locale, since they have it as `en-US` it breaks if we don't change it to `mc-US`, this does not happen in the json files. So here I'm just checking in a general way that if their locale is different from how we want to save it, that we change that key in the file.